src/monodroid/jni/monodroid-networkinfo.c

```diff
@@ -77,9 +79,9 @@ _monodroid_get_network_interface_state (const char *ifname, mono_bool *is_up, mo
 		goto leave;
 	}
 
-	JNIEnv *env = get_jnienv ();
+	env = get_jnienv ();
 	jstring NetworkInterface_nameArg = (*env)->NewStringUTF (env, ifname);
-	jobject networkInterface = (*env)->CallStaticObjectMethod (env, NetworkInterface_class, NetworkInterface_getByName, NetworkInterface_nameArg);
+	networkInterface = (*env)->CallStaticObjectMethod (env, NetworkInterface_class, NetworkInterface_getByName, NetworkInterface_nameArg);
```

Comment:
We should also call `JNIEnv::ExceptionClear()` (and maybe `JNIEnv::ExceptionOccurred()`?), in case a pending exception was raised by this call.  [`NetworkInterface.getByName(String)`](https://docs.oracle.com/javase/7/docs/api/java/net/NetworkInterface.html#getByName(java.lang.String)) declares at least two throwable exception types.